### DISCUSSION TITLE
Calculating kinship coefficients in multi-generational pedigrees

### DISCUSSION THREAD

**Discussion by vaqm2** [tskit-dev/tsinfer#787] [Q&A] [answered]
Hi All,

I have a genealogical dataset of ~ 2.5 million individuals in a format where each individual is in a row with the mother and father IDs in columns along with the sex and date of birth.  I wish to calculate the pairwise kinship coefficient between individuals within this multi-generational pedigree and for example, list all relatives for an individual that surpass a certain kinship coefficient in an efficient manner.  This seems doable with the tskit library but I wanted to check if there is a similar implementation or tutorial for the methods that can get me started.  I appreciate the help.

Thanks,
Vivek.

**Comment by hyanwong** [comment]
Hi @vaqm2 - if I'm reading you correctly, this is purely a problem involving pedigrees, rather than genetic genealogies. Is that right? There is support in tskit (not tsinfer) for pedigrees, but I'm not sure it's been optimised much. @benjeffery: can we transfer this to the tskit discussions forum?

**Reply by vaqm2** [reply]
Hi, yes. This is purely pedigrees where I can build a tree using the parental information and then connect to siblings, cousins etc and finally wish to compute the kinship coefficients between the individuals.  I'd appreciate if this can be moved to the appropriate forum.

**Comment by jeromekelleher** [comment, marked as answer]
We don't have methods for computing these types of stats in tskit @vaqm2 I'm afraid. However, the sgkit library has a [pedigree_kinship](https://pystatgen.github.io/sgkit/latest/generated/sgkit.pedigree_kinship.html) function which may compute what you need?

**Reply by vaqm2** [reply]
Thank you, I'll take a look at sgkit.

**Reply by timothymillar** [reply]
@vaqm2 please feel free to tag me in a discussion or issue on the sgkit page if you have any questions. We currently just support computing a square kinship matrix for a full pedigree. But if I can understand a bit more about your use-case, then I may be able to make some suggestions.
